src/frontend/src/lib/components/views/RecoveryPhraseInput.svelte

```diff
@@ -0,0 +1,160 @@
+<script lang="ts">
+  import Tooltip from "$lib/components/ui/Tooltip.svelte";
+  import { InfoIcon } from "@lucide/svelte";
+  import { t } from "$lib//stores/locale.store";
+  import { onMount } from "svelte";
+
+  interface Props {
+    value: string[];
+    showAll?: boolean;
+    disabled?: boolean;
+  }
+
+  let {
+    value = $bindable(),
+    showAll = false,
+    disabled = false,
+  }: Props = $props();
+
+  let words = $derived(Array.from(value));
+  let wrapperRef = $state<HTMLDivElement>();
+  let dictionary = $state<string[]>();
+
+  const handleKeyDown = (event: KeyboardEvent, index: number) => {
+    if (event.code === "Backspace" && words[index].length === 0) {
+      wrapperRef?.querySelectorAll("input")[index - 1]?.focus();
+      event.preventDefault();
+    }
+    if (event.code === "Space" || event.code === "Enter") {
+      wrapperRef?.querySelectorAll("input")[index + 1]?.focus();
+      event.preventDefault();
+    }
+  };
+  const handlePaste = (event: ClipboardEvent, index: number) => {
+    const clipboard = (event.clipboardData?.getData("text/plain") ?? "")
+      .trim()
+      .split(/\s/)
+      .map((word) => word.toLowerCase().replace(/[^a-z]/g, ""));
+    if (clipboard.length > 0) {
+      clipboard.forEach((word, i) => (words[index + i] = word));
+      words = [...words];
+      event.preventDefault();
+      wrapperRef
+        ?.querySelectorAll("input")
+        [index + clipboard.length - 1]?.focus();
+    }
+  };
+
+  onMount(() => {
+    // Lazy load dictionary so this component doesn't include it in the bundle eagerly
+    import("bip39").then((bip39) => (dictionary = bip39.wordlists.english));
+    // Focus on first empty input
+    wrapperRef
+      ?.querySelectorAll("input")
+      [words.findIndex((word) => word.length === 0)]?.focus();
+  });
+
+  $effect(() => {
```

Comment:
I can't easily understand this function's intention. 

It feels like we're adding a lot of inline functions that do clever things, and have zero tests and zero doc strings for their intended behavior. Am I missing something?